src/redux/missions/missions.js

```diff
@@ -0,0 +1,70 @@
+import axios from 'axios';
+
+const GET_MISSONS = 'spacetravel/missions/GET_MISSIONS';
+// const MISSIONS_REQUEST = 'spacetravel/missions/MISSIONS_REQUEST';
+// const MISSIONS_SUCCESS = 'spacetravel/missions/MISSIONS_SUCCESS';
+const BOOK_MISSION = 'spacetravel/missions/BOOK_MISSION';
+const LEAVE_MISSION = 'spacetravel/missions/LEAVE_MISSION';
+
+export const getMissions = () => async (dispatch) => {
+  const { data } = await axios.get('https://api.spacexdata.com/v3/missions');
+  const result = [];
+  data.forEach((item) => {
+    const newMission = {
+      id: item.mission_id,
+      name: item.mission_name,
+      reserved: false,
+      description: item.description,
+    };
+    result.push(newMission);
+  });
+
+  dispatch({ type: GET_MISSONS, payload: result });
+};
+
+export const bookMission = (payload) => (dispatch) => {
+  dispatch({ type: BOOK_MISSION, payload });
+};
+
+export const leaveMission = (payload) => (dispatch) => {
+  dispatch({ type: LEAVE_MISSION, payload });
+};
```

Comment:
- [ ] Here you are using an unnecessary async function for those types of actions (`synchronous actions`). Note that these actions are not blocking, the payload that they are sending to the reducer function is already there, so you don't need to make them asynchronous. They will be dispatched in your component, e.g `dispatch(leaveMission(missionId)`

- [ ] Let's keep them simple like this
```
export const bookMission = (payload) => ({ type: BOOK_MISSION, payload });

export const leaveMission = (payload) => ({ type: LEAVE_MISSION, payload });
```